tutorials/onchain-governance/fast-track-gov-proposal.md

```diff
@@ -0,0 +1,224 @@
+
+!!!note
+    In this tutorial, the focus will be on the main steps and core logic within the main function. For clarity and conciseness, the implementation details of individual functions will be available in expandable tabs below each section. At the end of the tutorial, you'll find the complete code for reference.
+
+### Spin Up the Polkadot Fork
+
+To set up your Polkadot fork using Chopsticks, open a new terminal window and run the following command:
+
+```bash
+npx @acala-network/chopsticks --config=polkadot
+```
+
+This command will start a local fork of the Polkadot network accesible at `ws://localhost:8000`. Keep this terminal window open and running throughout your testing process.
+
+Once your forked network is up and running, you can proceed with the following steps.
+
+### Set Up Dependencies and Structure
+
+Begin by adding the necessary imports and a basic structure to the `test-proposal.ts` file:
+
+```typescript
+--8<-- 'code/tutorials/onchain-governance/fast-track-gov-proposal/test-proposal.ts:imports'
+
+const main = async () => {
+  // The code will be added here
+
+  process.exit(0);
+}
+
+--8<-- 'code/tutorials/onchain-governance/fast-track-gov-proposal/test-proposal.ts:try-catch-block'
+```
+
+This structure provides the foundation for your script. It imports all the necessary dependencies and sets up a main function that will contain the core logic of your proposal submission process.
+
+### Connect to the Forked Chain
+
+Create a `connectToFork` function outside the `main` function to connect your locally forked chain to the Polkadot.js API:
+
+```typescript
+    --8<-- 'code/tutorials/onchain-governance/fast-track-gov-proposal/test-proposal.ts:connectToFork'
+```
+
+Inside the `main` function, add the code to establish a connection to your local Polkadot fork:
+
+```typescript hl_lines="2-3"
+--8<-- 'code/tutorials/onchain-governance/fast-track-gov-proposal/test-proposal.ts:226:229'
+  ...
+}
+```
+
+### Create and Submit the Proposal
+
+Create a `generateProposal` function responsible for preparing and submitting the on-chain proposal. The function implements a workflow that involves:
+
+1. Configuring a keyring for the Alice account to facilitate transaction submission. When using Chopsticks, this development account is pre-funded to execute all necessary actions
+
+2. Executing a batch transaction that comprises three critical operations:
+    1. `preimage.notePreimage` - registers a [preimage](/polkadot-protocol/glossary#preimage){target=\_blank} using the selected call
+
+        !!!note
+            The preimage hash is simply the hash of the proposal to be enacted. The on-chain proposals do not require the entire image of extrinsics and data (for instance the Wasm code, in case of upgrades) to be submitted, but would rather just need that image's hash. That preimage can be submitted and stored on-chain against the hash later, upon the proposal's dispatch.
```

Comment:
```suggestion
            The preimage hash is simply the hash of the proposal to be enacted. The on-chain proposals do not require the entire image of extrinsics and data (for instance, the Wasm code, in case of upgrades) to be submitted but would need that image's hash. That preimage can be submitted and stored on-chain against the hash later upon the proposal's dispatch.
```